tests/kernel/stack fails to build on nios2 with new SDK 0.10.0-rc3

**Describe the bug**
tests/kernel/stack build failure on qemu_nios2 with new SDK


**To Reproduce**

using the new SDK, build tests/kernel/stack for nios2

**Expected behavior**
it builds cleanly

**Screenshots or console output**
```
opt/zephyr-sdk-0.10.0-rc2/nios2-zephyr-elf/bin/../lib/gcc/nios2-zephyr-elf/8.2.0/../../../../nios2-zephyr-elf/bin/ld: ../app/libapp.a(test_stack_contexts.c.obj): in function `tstack_pop':
/home/galak/git/zephyr/tests/kernel/stack/stack_api/src/test_stack_contexts.c:31: warning: unable to reach (null) (at 0x004002f4) from the global pointer (at 0x004082fc) because the offset (-32776) is out of the allowed range, -32678 to 32767
```

**Environment (please complete the following information):**
 - Toolchain SDK 0.10.0-rc3 (https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v0.10.0-rc3/zephyr-sdk-0.10.0-rc3-setup.run)

--- a/tests/kernel/stack/stack_api/src/test_stack_contexts.c
+++ b/tests/kernel/stack/stack_api/src/test_stack_contexts.c
@@ -7,7 +7,7 @@
 #include <ztest.h>
 #include <irq_offload.h>
 #define STACK_SIZE 512
-#define STACK_LEN 2
+#define STACK_LEN 4
 
 /**TESTPOINT: init via K_STACK_DEFINE*/
 K_STACK_DEFINE(kstack, STACK_LEN);

